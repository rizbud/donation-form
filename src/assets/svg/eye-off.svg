<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
  <path d="M19.8 22.6L15.6 18.45C15.0167 18.6333 14.4293 18.771 13.838 18.863C13.246 18.9543 12.6333 19 12 19C9.48333 19 7.24167 18.3043 5.275 16.913C3.30833 15.521 1.88333 13.7167 1 11.5C1.35 10.6167 1.79167 9.79567 2.325 9.037C2.85833 8.279 3.46667 7.6 4.15 7L1.4 4.2L2.8 2.8L21.2 21.2L19.8 22.6ZM12 16C12.1833 16 12.354 15.9917 12.512 15.975C12.6707 15.9583 12.8417 15.925 13.025 15.875L7.625 10.475C7.575 10.6583 7.54167 10.8293 7.525 10.988C7.50833 11.146 7.5 11.3167 7.5 11.5C7.5 12.75 7.93733 13.8127 8.812 14.688C9.68733 15.5627 10.75 16 12 16ZM19.3 16.45L16.125 13.3C16.2417 13.0167 16.3333 12.7293 16.4 12.438C16.4667 12.146 16.5 11.8333 16.5 11.5C16.5 10.25 16.0627 9.18733 15.188 8.312C14.3127 7.43733 13.25 7 12 7C11.6667 7 11.354 7.03333 11.062 7.1C10.7707 7.16667 10.4833 7.26667 10.2 7.4L7.65 4.85C8.33333 4.56667 9.03333 4.354 9.75 4.212C10.4667 4.07067 11.2167 4 12 4C14.5167 4 16.7583 4.69567 18.725 6.087C20.6917 7.479 22.1167 9.28333 23 11.5C22.6167 12.4833 22.1127 13.396 21.488 14.238C20.8627 15.0793 20.1333 15.8167 19.3 16.45ZM14.675 11.85L11.675 8.85C12.1417 8.76667 12.571 8.804 12.963 8.962C13.3543 9.12067 13.6917 9.35 13.975 9.65C14.2583 9.95 14.4627 10.296 14.588 10.688C14.7127 11.0793 14.7417 11.4667 14.675 11.85Z" fill="currentColor"/>
</svg>