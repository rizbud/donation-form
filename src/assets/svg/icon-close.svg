<svg width="18" height="28" viewBox="0 0 18 28" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_10_1505)">
<path d="M17.4713 19.7639C18.1744 20.4475 18.1744 21.5576 17.4713 22.2412C16.7682 22.9248 15.6263 22.9248 14.9232 22.2412L9.00007 16.4771L3.07132 22.2357C2.36819 22.9193 1.22632 22.9193 0.523193 22.2357C-0.179932 21.5521 -0.179932 20.442 0.523193 19.7584L6.45194 13.9998L0.528818 8.23574C-0.174307 7.55215 -0.174307 6.44199 0.528818 5.7584C1.23194 5.0748 2.37382 5.0748 3.07694 5.7584L9.00007 11.5225L14.9288 5.76387C15.6319 5.08027 16.7738 5.08027 17.4769 5.76387C18.1801 6.44746 18.1801 7.55762 17.4769 8.24121L11.5482 13.9998L17.4713 19.7639Z" fill="black"/>
</g>
<defs>
<clipPath id="clip0_10_1505">
<rect width="18" height="28" fill="white" transform="matrix(1 0 0 -1 0 28)"/>
</clipPath>
</defs>
</svg>
