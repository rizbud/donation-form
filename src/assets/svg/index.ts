export { default as EyeOff } from "./eye-off.svg";
export { default as Eye } from "./eye.svg";
export { default as LoadingAnimation } from "./loading-animation.svg";
