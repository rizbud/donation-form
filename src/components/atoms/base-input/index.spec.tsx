import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import userEvent from "@testing-library/user-event";

import { BaseInput } from ".";

import { Eye } from "~/assets/svg";

describe("BaseInput", () => {
  it("should render a text input", () => {
    render(<BaseInput label="Label" type="text" />);

    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("should render a hidden text input", () => {
    render(<BaseInput label="Label" hidden />);

    expect(screen.getByLabelText("Label")).toBeInTheDocument();
  });

  it("should render a hidden text input", () => {
    render(<BaseInput label="Label" type="hidden" />);

    expect(screen.getByLabelText("Label")).toBeInTheDocument();
  });

  it("should render a email input", () => {
    render(<BaseInput label="Email" type="email" />);

    expect(screen.getByLabelText("Email")).toBeInTheDocument();
  });

  it("should render a text input with error message", () => {
    render(<BaseInput label="Label" type="text" error="Error" />);

    expect(screen.getByRole("textbox")).toBeInTheDocument();
    expect(screen.getByText("Error")).toBeInTheDocument();
  });

  it("should render a password input with action icon eye and onClick event", async () => {
    const onClick = jest.fn();

    render(
      <BaseInput
        label="Password"
        type="password"
        action={{ icon: Eye, onClick }}
      />,
    );

    expect(screen.getByLabelText("Password")).toBeInTheDocument();

    await userEvent.click(screen.getByRole("button"));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});

//  Snapshots
describe("BaseInput", () => {
  it("should render a text input", () => {
    const { container } = render(<BaseInput label="Label" type="text" />);

    expect(container).toMatchSnapshot();
  });

  it("should render a hidden text input", () => {
    const { container } = render(<BaseInput label="Label" hidden />);

    expect(container).toMatchSnapshot();
  });

  it("should render a hidden text input", () => {
    const { container } = render(<BaseInput label="Label" type="hidden" />);

    expect(container).toMatchSnapshot();
  });

  it("should render a email input", () => {
    const { container } = render(<BaseInput label="Email" type="email" />);

    expect(container).toMatchSnapshot();
  });

  it("should render a text input with error message", () => {
    const { container } = render(
      <BaseInput label="Label" type="text" error="Error" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render a password input with action icon eye and onClick event", async () => {
    const onClick = jest.fn();

    const { container } = render(
      <BaseInput
        label="Password"
        type="password"
        action={{ icon: Eye, onClick }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
