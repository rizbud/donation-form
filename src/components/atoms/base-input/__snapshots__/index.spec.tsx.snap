// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInput should render a email input 1`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Email
    </label>
    <div
      class="relative flex items-center"
    >
      <input
        aria-invalid="false"
        aria-label="Email"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
        type="email"
      />
    </div>
  </div>
</div>
`;

exports[`BaseInput should render a hidden text input 1`] = `
<div>
  <div
    class="hidden mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Label
    </label>
    <div
      class="relative flex items-center"
    >
      <input
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
        hidden=""
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`BaseInput should render a hidden text input 2`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Label
    </label>
    <div
      class="relative flex items-center hidden"
    >
      <input
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
        type="hidden"
      />
    </div>
  </div>
</div>
`;

exports[`BaseInput should render a password input with action icon eye and onClick event 1`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Password
    </label>
    <div
      class="relative flex items-center"
    >
      <input
        aria-invalid="false"
        aria-label="Password"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary pr-10"
        type="password"
      />
      <button
        class="absolute inset-y-0 right-0 flex items-center pr-3"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="text-secondary h-6 w-6"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`BaseInput should render a text input 1`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Label
    </label>
    <div
      class="relative flex items-center"
    >
      <input
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`BaseInput should render a text input with error message 1`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Label
    </label>
    <div
      class="relative flex items-center"
    >
      <input
        aria-invalid="true"
        aria-label="Label"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:border-red-500 enabled:focus:ring-red-500"
        type="text"
      />
    </div>
    <span
      class="text-sm mt-1 text-red-500"
    >
      Error
    </span>
  </div>
</div>
`;

exports[`BaseInput should render a text input with error message 2`] = `
<div>
  <div
    class="mb-6"
  >
    <label
      class="block text-sm font-semibold text-slate-800 mb-1"
    >
      Label
    </label>
    <div
      class="relative flex items-center"
    >
      <span
        class="absolute inset-y-0 left-0 flex items-center pb-px"
      >
        $
      </span>
      <input
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-secondary text-black border-secondary block w-full border-0 border-b px-0 py-1 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-0 enabled:focus:border-primary enabled:focus:ring-primary pl-4"
        type="text"
      />
    </div>
  </div>
</div>
`;
